components/org.wso2.carbon.siddhi.error.handler.core/src/main/java/org/wso2/carbon/siddhi/error/handler/core/execution/RePlayer.java

```diff
@@ -60,57 +60,114 @@ public static void rePlay(List<ErrorEntry> errorEntries) throws SiddhiErrorHandl
 
     private static void rePlay(ErrorEntry errorEntry) throws SiddhiErrorHandlerException, InterruptedException {
         SiddhiAppRuntime siddhiAppRuntime = SiddhiErrorHandlerDataHolder.getInstance()
-            .getSiddhiAppRuntimeService().getActiveSiddhiAppRuntimes().get(errorEntry.getSiddhiAppName());
-        if (siddhiAppRuntime != null) {
-            switch (errorEntry.getEventType()) {
-                case COMPLEX_EVENT:
-                    rePlayComplexEvent(errorEntry, siddhiAppRuntime);
-                    break;
-                case EVENT:
-                    rePlayEvent(errorEntry, siddhiAppRuntime);
-                    break;
-                case EVENT_ARRAY:
-                    rePlayEventArray(errorEntry, siddhiAppRuntime);
-                    break;
-                case EVENT_LIST:
-                    rePlayEventList(errorEntry, siddhiAppRuntime);
-                    break;
-                case PAYLOAD_STRING:
-                    rePlayPayloadString(errorEntry, siddhiAppRuntime);
-                    break;
-                default:
-                    // Ideally we won't reach here
```

Comment:
Let's add the following case here:
```
case REPLAYABLE_TABLE_RECORD:
    rePlayComplexEventChunk(errorEntry, siddhiAppRuntime);
    break;
```
and let's remove the **else** part of `if (errorEntry.getErrorType() == ErrorType.MAPPING || errorEntry.getErrorType() == ErrorType.TRANSPORT) {`